pkg/insightstore/filestore.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightstore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pipe-cd/pipe/pkg/filestore"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+type insightFileStore struct {
+	filestore filestore.Store
+}
+
+// GetReports returns data as report.
+func (f *insightFileStore) GetReports(
+	ctx context.Context,
+	projectID string,
+	appID string,
+	metricsKind model.InsightMetricsKind,
+	step model.InsightStep,
+	from time.Time,
+	dataPointCount int) ([]Report, error) {
+	from = formatFrom(from, step)
+
+	paths := searchFilePaths(projectID, appID, from, dataPointCount, metricsKind, step)
+	var reports []Report
+	for _, p := range paths {
+		r, err := f.getReport(ctx, p, metricsKind)
+		if err != nil {
+			return nil, err
+		}
+
+		reports = append(reports, r)
+
+		dataPointCount = dataPointCount - r.DataCount(step)
+
+		nextMonth := time.Date(from.Year(), from.Month()+1, 1, 0, 0, 0, 0, time.UTC)
+		from = formatFrom(nextMonth, step)
+		if step == model.InsightStep_WEEKLY && from.Month() != nextMonth.Month() {
+			from = from.AddDate(0, 0, 7)
+		}
```

Comment:
Sorry, what is the purpose of this part?

Reply:
Here we should specify the first Sunday in the next month, but, because formatFrom returns the Sunday of the week that includes the day `nextMonth `, it is possible that the last Sunday of the previous month will be specified for the date `from`.

Reply:
Sorry that my question was unclear. As I see, the block from line 52 to line 58 does not change any variables or make any effect on other parts. It just updates the `from` variable but that variable is used once before entering the loop. 
Sorry if I missed something.

Reply:
Oops, these processes should have been implemented in the list function... ; ;